src/o365/graph/commands/teams/teams-add.ts

```diff
@@ -0,0 +1,197 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../GraphCommand';
+import * as request from 'request-promise-native';
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  groupId?: string;
+  name?: string;
+  description?: string;
+}
+
+class GraphTeamsAddCommand extends GraphCommand {
+  public get name(): string {
+    return `${commands.TEAMS_ADD}`;
+  }
+
+  public get description(): string {
+    return 'Add a new Microsoft Teams team.';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    return telemetryProps;
```

Comment:
We should be tracking all optional arguments here.

Reply:
Included all the optional parameters.